io/io/src/TGenCollectionProxy.cxx

```diff
@@ -325,13 +325,12 @@ TGenCollectionProxy::Value::Value(const std::string& inside_type, Bool_t silent,
    fKind = kNoType_t;
 
    // Let's treat the unique_ptr case
-   bool nameChanged = false;
-   std::string intype = TClassEdit::GetNameForIO(inside.c_str(), TClassEdit::EModType::kNone, &nameChanged);
+   std::string intype = TClassEdit::GetNameForIO(inside, TClassEdit::EModType::kNone);
 
-   bool isPointer = nameChanged; // unique_ptr is considered a pointer
+   bool isPointer = false;
    // The incoming name is normalized (it comes from splitting the name of a TClass),
    // so all we need to do is drop the last trailing star (if any) and record that information.
-   if (!nameChanged && intype[intype.length()-1] == '*') {
+   if (intype[intype.length()-1] == '*') {
```

Comment:
Also, in `TStreamerElement.cxx` the same logic is implemented with a `while`, is there a reason for the asymmetry?

Reply:
Yes, it is not quite the same logic.  In TStreamerElement, we simply try to retrieve the underlying type while here we issue a Warning as soon as we get two 2 levels.